genai-perf/genai_perf/inputs/retrievers/synthetic_audio_generator.py

```diff
@@ -0,0 +1,141 @@
+        
+        Args:
+            mean: Mean value for the normal distribution
+            stddev: Standard deviation for the normal distribution
+            min_value: Minimum acceptable value
+            
+        Returns:
+            A positive sample from the normal distribution
+        """
+        if mean < min_value:
+            raise ValueError(f"Mean value ({mean}) must be greater than min_value ({min_value})")
+        
+        while True:
+            sample = np.random.normal(mean, stddev)
+            if sample >= min_value:
+                return sample
+
+    @staticmethod
+    def _validate_sampling_rate(sampling_rate: int, output_format: AudioFormat) -> None:
+        """
+        Validate sampling rate for the given output format.
+        
+        Args:
+            sampling_rate: Sampling rate in Hz
+            output_format: Output audio format
+            
+        Raises:
+            ValueError: If sampling rate is not supported for the given format
+        """
+        if output_format.lower() == "mp3" and sampling_rate not in MP3_SUPPORTED_SAMPLE_RATES:
+            supported_rates = sorted(MP3_SUPPORTED_SAMPLE_RATES)
+            raise ValueError(
+                f"MP3 format only supports the following sample rates (in Hz): {supported_rates}. "
+                f"Got {sampling_rate} Hz. Please choose a supported rate from the list."
+            )
+
+    @staticmethod
+    def create_synthetic_audio(
+        audio_length_mean: float,
+        audio_length_stddev: float,
+        sampling_rates_khz: List[int],
+        bit_depths: List[int],
+        output_format: AudioFormat = "wav",
+    ) -> Dict[str, Dict[str, str]]:
+        """
+        Generate synthetic audio data with specified parameters.
+        
+        Args:
+            audio_length_mean: Mean audio length in seconds (must be >= 0.1)
+            audio_length_stddev: Standard deviation of audio length in seconds
+            sampling_rates_khz: List of allowed sampling rates in kHz
+            bit_depths: List of allowed bit depths in bits
+            output_format: Output audio format, either "wav" or "mp3"
+            
+        Returns:
+            Dictionary compatible with OpenAI API input_audio format:
+            {
+                "type": "input_audio",
+                "input_audio": {
+                    "data": "<base64-encoded audio>",
+                    "format": "<wav or mp3>"
+                }
+            }
+            
+        Raises:
+            ValueError: If sampling rate is not supported for the given format
+        """
+        # Sample audio length (in seconds) using rejection sampling
+        audio_length = SyntheticAudioGenerator._sample_positive_normal(
+            audio_length_mean, audio_length_stddev
+        )
+        
+        # Randomly select sampling rate and bit depth
+        sampling_rate = int(np.random.choice(sampling_rates_khz) * 1000)  # Convert kHz to Hz
```

Comment:
Should sampling rate be truncated to be integers?  I think one of the common sampling rates is float (e.g. 44.1 kHz).

Reply:
Nvm, for some reason, my brain just decided to ignore multiplying by 1000 part which prevents the truncation issue. Current way looks good.